.github/workflows/DiffTool-CI.yml

```diff
@@ -0,0 +1,133 @@
+name: CI/CD for DiffTool
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:  # Allows manual triggering
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    env:
+      JAVA_VERSION: '21'
+      CONFIG_DIR: '.ci-config'
+      DIFFTOOL_JAR: "diff-java-tool.jar"
+      PATCH_DIFF_TOOL_VERSION: "0.1-SNAPSHOT"
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          clean: true
+          fetch-depth: 1
+
+      - name: Set up JDK ${{ env.JAVA_VERSION }}
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: ${{ env.JAVA_VERSION }}
+          cache: 'gradle'
+
+      - name: Cache Maven Dependencies
+        uses: actions/cache@v4
+        with:
+          path: ~/.m2/repository
+          key: maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            maven-
+
+      - name: Build with Gradle
+        run: |
+          cd diff-java-tool
+          ./gradlew clean build
+
+      - name: Copy DiffTool JAR to workspace
+        run: |
+          cp diff-java-tool/build/libs/diff-java-tool-*-all.jar "${{ github.workspace }}/${{ env.DIFFTOOL_JAR }}"
+
+      - name: Download patch-diff-report-tool JAR
+        uses: robinraju/release-downloader@v1.8
+        with:
+          repository: "checkstyle/contribution"
+          tag: "patch-diff-report-tool-${{ env.PATCH_DIFF_TOOL_VERSION }}"
+          fileName: >-
+            patch-diff-report-tool-${{ env.PATCH_DIFF_TOOL_VERSION }}-jar-with-dependencies.jar
+          out-file-path: "${{ github.workspace }}"
+
+      - name: Clone Checkstyle Repository
+        run: |
+          git clone https://github.com/checkstyle/checkstyle.git "${{ github.workspace }}/checkstyle"
```

Comment:
can we reuse action for this ?
https://github.com/actions/checkout?tab=readme-ov-file#checkout-multiple-repos-side-by-side

Reply:
done